src/unitquaternion.jl

```diff
@@ -38,6 +38,20 @@ struct QuatRotation{T} <: Rotation{3,T}
     QuatRotation{T}(q::QuatRotation) where T = new{T}(q.q)
 end
 
+function Base.getproperty(q::QuatRotation, f::Symbol)
+    if f == :w
```

Comment:
Thanks for the review!

>So this just restores the previous names for these fields?

Additionally, this PR renames `_pure_quaternion` to `pure_quaternion`.
In, `v1.0.1`, there was [`pure_quaternion` function](https://github.com/JuliaGeometry/Rotations.jl/blob/v1.0.4/src/unitquaternion.jl#L233) which returns `UnitQuaternion`, but the function does NOT return *unit* quaternion.

I thought this may be a bug, and I renamed this function to [`_pure_quaternion`](https://github.com/JuliaGeometry/Rotations.jl/blob/v1.1.0/src/unitquaternion.jl#L254) which returns `Quaternion` in `v1.1.0`.

This causes [an issue](https://github.com/RoboticExplorationLab/RobotDynamics.jl/issues/12), and this seems easy to fix by just reverting the name.

I'm not familiar with the original expected behavior and usage of `pure_quaternion`, but it seems to be fine as far as RobotDynamics.jl is concerned.